src/basic/time-util.c

```diff
@@ -55,6 +56,17 @@ usec_t now(clockid_t clock_id) {
         return timespec_load(&ts);
 }
 
+usec_t epoch_or_now(clockid_t clock_id) {
+        int r;
+        unsigned long long epoch;
+
+        r = getenv_llu_secure("SOURCE_DATE_EPOCH", 10, &epoch);
+        if (r == 0)
+                return (usec_t) epoch * USEC_PER_SEC;
+
+        return now(clock_id);
```

Comment:
SOURCE_DATE_EPOCH is wallclock time. hence this function might either return time of the wallclock or time as specified in `clock_id`... That's broken. Just drop the parameter, and hard code CLOCK_REALTIME here. No other argument makes sense.